### PR TITLE
Integrated React with Django-Rest Framework to fetch backend data and display it

### PR DESCRIPTION
**Pull request by musanaeem**
Calling APIs to retrieve data from DRF backend, I created frontend pages for BlogList, Blog and Bio details. CRUD functionalities are to be added in next PR.

- Navbar created and logout functionality taken to the Navbar and thus wrapper removed around the Homepage.
- Homepage logout button removed as a global navbar is being used for that purpose
- Links, useNavigate and useLocation used to redirect user to pages and pass/retrieve data 
- Links added instead of anchor tags so page doesn't re-render every time and only the data part is re-rendered.
- Blog request takes optional parameter to deal with both blogs page and single blog page requests.

Screenshots:

BlogListPage:
Screenshot 2022-10-21 at 1 46 28 PM

BlogPage:
Screenshot 2022-10-21 at 1 46 35 PM

BioPage:
Screenshot 2022-10-21 at 1 46 21 PM

### CODE REVIEW

File: src/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+function Navbar(props) {
+
+    const navigate = useNavigate();
+
+    const logoutUser = () => {
+        window.localStorage.setItem('blogSiteUserLoggedIn', false);
+        logoutRequest().then();
+        props.changeLoginState(false);
+        navigate('/login');
```
Comment (srehaider) [inline src/components/Navbar.js]: Please have a look at how promises work, you should only perform logout tasks on frontend when you receive a successful response from backend.

File: src/components/RegisterForm.js [hunk 2]
```diff
@@ -143,6 +143,7 @@ const RegisterForm = (props) => {
     }
     
     const isValidEmail = (emailValue) => {
+        // eslint-disable-next-line no-useless-escape
```
Comment (srehaider) [inline src/components/RegisterForm.js]: Learn about this rule, it is a valid concern.

File: src/components/Routing.js [hunk 3]
```diff
@@ -45,17 +49,24 @@ class Routing extends Component {
 
   render() {
     return (
+      <>
+      {this.state.isLoggedIn && <Navbar  changeLoginState = {this.updateLoginState}/>}
```
Comment (srehaider) [inline src/components/Routing.js]: NavBar always should be visible just change its content based on if a user is signed in or not.

File: src/components/bioStyle.css [hunk 4]
```diff
@@ -0,0 +1,93 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto+Slab&display=swap');
+
+.bioPage .user {
```
Comment (srehaider) [inline src/components/bioStyle.css]: Don't use compound selectors everywhere if two elements have different styles use different classes.

File: src/pages/BioPage.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import bioRequest from '../services/BioRequest';
+import '../components/bioStyle.css'
+
+const BioPage = (props) => {
+    const navigate = useNavigate();
+    const [bioData, setBioData] = useState('');
+    const method = 'GET';
+    const body = '';
+    
+    const fetchTasksOrRedirect = () => {
```
Comment (srehaider) [inline src/pages/BioPage.js]:
```suggestion
    const fetchBioOrRedirect = () => {
```

File: src/pages/BlogListPage.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import moment from 'moment';
+
+
+const BlogListPage = (props) => {
+
+    const navigate = useNavigate();
+    const [blogData, setBlogData] = useState('');
+    
+    
+    const fetchTasksOrRedirect = () => {
```
Comment (srehaider) [inline src/pages/BlogListPage.js]:
```suggestion
    const fetchBlogsOrRedirect = () => {
```

**Review by srehaider — CHANGES_REQUESTED** [review]
If you are using ESLint for linting please add linting packages and linting rules too, so all collaborators can use them.

File: src/components/Routing.js (hunk 3 above)
Comment (musanaeem) [reply src/components/Routing.js]: Done

File: src/pages/BioPage.js (hunk 5 above)
Comment (musanaeem) [reply src/pages/BioPage.js]: Done

File: src/pages/BlogListPage.js (hunk 6 above)
Comment (musanaeem) [reply src/pages/BlogListPage.js]: Done

File: src/components/Navbar.js (hunk 1 above)
Comment (musanaeem) [reply src/components/Navbar.js]: Done

File: src/components/RegisterForm.js (hunk 2 above)
Comment (musanaeem) [reply src/components/RegisterForm.js]: Done

File: src/components/bioStyle.css (hunk 4 above)
Comment (musanaeem) [reply src/components/bioStyle.css]: Done

File: src/components/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            </div>
+        ) : (<div className='nav'>
+                <Link to='/home' className='site-title'> Arbisoft Blogs </Link>
+                    <ul>
+                        <li>
+                            <Link to='/login' className='signIn-btn'> Sign In </Link>
+                        </li>
+                    </ul>
+            </div>)
+    }
```
Comment (srehaider) [inline src/components/Navbar.js]:
```suggestion
        <div className='nav'>
            <Link to='/home' className='site-title'> Arbisoft Blogs </Link>
            <ul>
                {props.loginState ?
                    (
                        <li>
                            <Link to='/bio'> Bio </Link>
                        </li>
                        <li>
                            <Link to='/blog'> Blogs </Link>
                        </li>
                        <li className='logout-li' onClick={logoutUser}>
                            Logout 
                        </li>
                    ) : (
                        <li>
                            <Link to='/login' className='signIn-btn'> Sign In </Link>
                        </li>
                    )
                }
            </ul>
        </div>
```

File: src/components/RecordContainer.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+  return (
+    <div className={getClassName()}>
+        <div className="labels">
+            {props.children}
+        </div> 
+    </div>
+  )
+}
```
Comment (srehaider) [inline src/components/RecordContainer.js]:
```suggestion
const RecordContainer = (props) => {
  return (
    <div className={`${props.className} record`}>
        <div className="labels">
            {props.children}
        </div> 
    </div>
  )
}
```

**Review by srehaider — APPROVED** [review]
Two smaller improvements can be made after these changes feel to merge.

File: src/components/Navbar.js (hunk 7 above)
Comment (musanaeem) [reply src/components/Navbar.js]: Done

File: src/components/RecordContainer.js (hunk 8 above)
Comment (musanaeem) [reply src/components/RecordContainer.js]: Done

**Pull request merged**